contributingGuides/STYLE.md

```diff
@@ -10,13 +69,514 @@ We use Prettier to automatically style our code.
+
+When an exception is granted, link the relevant Slack conversation in your PR. Suppress ESLint or TypeScript warnings/errors with comments if necessary.
+
+This rule will apply until the migration is done. After the migration, discussion on granting exception can happen inside the PR page and doesn't need take place in the Slack channel.
+
+### Communication Items
+
+> Comment in the `#expensify-open-source` Slack channel if any of the following situations are encountered. Each comment should be prefixed with `TS ATTENTION:`. Internal engineers will access each situation and prescribe solutions to each case. Internal engineers should refer to general solutions to each situation that follows each list item.
+
+- I think types definitions in a third party library or JavaScript's built-in module are incomplete or incorrect
+
+When the library indeed contains incorrect or missing type definitions and it cannot be updated, use module augmentation to correct them. All module augmentation code should be contained in `/src/types/modules/*.d.ts`, each library as a separate file.
+
+```ts
+// external-library-name.d.ts
+
+declare module "external-library-name" {
+    interface LibraryComponentProps {
+        // Add or modify typings
+        additionalProp: string;
+    }
+}
+```
+
+### Other Expensify Resources on TypeScript
+
+- [Expensify TypeScript React Native CheatSheet](./TS_CHEATSHEET.md)
+
 ## Naming Conventions
 
+### Type names
+
+  - Use PascalCase for type names. eslint: [`@typescript-eslint/naming-convention`](https://typescript-eslint.io/rules/naming-convention/)
+
+    ```ts
+    // BAD
+    type foo = ...;
+    type BAR = ...;
+
+    // GOOD
+    type Foo = ...;
+    type Bar = ...;
+    ```
+
+  - Do not postfix type aliases with `Type`.
+
+    ```ts
+    // BAD
+    type PersonType = ...;
+
+    // GOOD
+    type Person = ...;
+    ```
+
+  - Use singular name for union types.
+
+    ```ts
+    // BAD
+    type Colors = "red" | "blue" | "green";
+
+    // GOOD
+    type Color = "red" | "blue" | "green";
+    ```
+
+  - Use `{ComponentName}Props` pattern for prop types.
+
+    ```ts
+    // BAD
+    type Props = {
+        // component's props
+    };
+
+    function MyComponent({}: Props) {
+        // component's code
+    }
+
+    // GOOD
+    type MyComponentProps = {
+        // component's props
+    };
+
+    function MyComponent({}: MyComponentProps) {
+        // component's code
+    }
+    ```
+
+  - Use {ComponentName}Handle for custon ref handle types.
+
+    ```tsx
+    // BAD
+    type MyComponentRef = {
+        onPressed: () => void;
+    };
+
+    // GOOD
+    type MyComponentHandle = {
```

Comment:
We may need to remove this rule or adjust the code as the rule isn't really used
![image](https://github.com/Expensify/App/assets/46095609/033336d3-4764-4260-b93c-a9b4f9d32f6a)
vs
![image](https://github.com/Expensify/App/assets/46095609/31e6806b-268f-4e5f-bdcc-d838c9c4056d)